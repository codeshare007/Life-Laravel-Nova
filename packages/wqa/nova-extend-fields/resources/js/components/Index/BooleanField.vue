<template>
  <div class="editMode" v-if="editMode">
    <ToggleCheckbox :field="field" :resource-row-id="resourceRowId" @input="toggleMode"/>
  </div>
  <div v-else>
    <div class="text-center">
        <span
            class="inline-block rounded-full w-2 h-2"
            :class="{'bg-success': field.value, 'bg-danger': !field.value}" />
    </div>
  </div>
</template>

<script>
  import ToggleCheckbox from '../tailwind/TailwindToogle'
  export default {

    props: ['resourceRowId', 'resourceName', 'field', 'editMode'],

    components: {
      ToggleCheckbox,
    },

    methods: {
      toggleMode(status) {
        // this.updateQueryString({
        //   ['editMode']: status
        // })
      }
    }
  }
</script>
