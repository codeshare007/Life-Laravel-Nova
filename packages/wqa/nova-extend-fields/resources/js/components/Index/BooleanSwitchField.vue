<template>
  <div class="editMode" v-if="editMode">
    <component
      :is="'form-' + field.component"
      :resource-name="resourceName"
      :field="field"
      :resourceRowId="resourceRowId"
      :showWrapper="false"
    />
  </div>
  <div v-else class="text-center">
    <span
      class="inline-block rounded-full w-4 h-4"
      :class="{'bg-success': field.value, 'bg-danger': !field.value}"
    />
  </div>
</template>

<script>
  export default {
    props: ['resourceName', 'resourceRowId', 'field', 'editMode'],
  }
</script>
