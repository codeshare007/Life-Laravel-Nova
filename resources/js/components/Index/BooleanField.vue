<template>
    <div class="text-center">
        <span
            class="inline-block rounded-full w-2 h-2"
            :class="{'bg-success': field.value, 'bg-danger': !field.value}" />
    </div>
</template>

<script>
export default {
    props: ['resourceName', 'field'],
}
</script>
