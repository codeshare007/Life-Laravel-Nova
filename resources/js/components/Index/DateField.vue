<template>
    <span v-if="field.value" class="whitespace-no-wrap">{{ formattedDate }}</span>
    <span v-else>&mdash;</span>
</template>

<script>
export default {
    props: ['resourceName', 'field'],

    computed: {
        formattedDate() {
            if (this.field.format) {
                return moment(this.field.value).format(this.field.format)
            }

            return this.field.value
        },
    },
}
</script>
