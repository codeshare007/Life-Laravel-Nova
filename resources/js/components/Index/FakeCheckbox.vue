<template>
    <div class="checkbox select-none rounded">
        <svg class="block" v-if="checked" width="20" height="20" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><g id="Page-1" fill="none" fill-rule="evenodd"><g id="checkbox-on" fill-rule="nonzero"><g id="b-link" fill="var(--primary)"><rect id="b" width="20" height="20" rx="4"/></g><path id="Shape" fill="#FFF" d="M7.7 9.3c-.23477048-.3130273-.63054226-.46037132-1.01285927-.37708287-.38231702.08328846-.68093514.38190658-.7642236.7642236C5.83962868 10.0694577 5.9869727 10.4652295 6.3 10.7l2 2c.38884351.3811429 1.01115649.3811429 1.4 0l4-4c.3130273-.23477048.4603713-.63054226.3770829-1.01285927-.0832885-.38231702-.3819066-.68093514-.7642236-.7642236C12.9305423 6.83962868 12.5347705 6.9869727 12.3 7.3L9 10.58l-1.3-1.3v.02z"/></g></g></svg>
        <svg class="block" v-else width="20" height="20" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><g id="Page-1" fill="none" fill-rule="evenodd"><g id="checkbox-off"><g id="b-link" fill="#FFF" fill-rule="nonzero"><rect id="b" width="20" height="20" rx="4"/></g><rect id="Rectangle-path" width="19" height="19" x=".5" y=".5" stroke="#CCD4DB" rx="4"/></g></g></svg>
    </div>
</template>

<script>
export default {
    props: {
        checked: {
            default: false,
        },
    },
}
</script>
