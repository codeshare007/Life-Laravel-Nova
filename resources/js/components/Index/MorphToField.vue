<template>
    <router-link
        v-if="field.value"
        :to="{
            name: 'detail',
            params: {
                resourceName: field.resourceName,
                resourceId: field.morphToId
            }
        }"
        class="no-underline dim text-primary font-bold"
    >
        {{ field.resourceLabel }}: {{ field.value }}
    </router-link>

    <span v-else>&mdash;</span>
</template>

<script>
export default {
    props: ['resourceName', 'field'],
}
</script>
