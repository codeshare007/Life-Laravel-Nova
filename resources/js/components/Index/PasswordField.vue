<template>
    <span class="font-bold">
        &middot;
        &middot;
        &middot;
        &middot;
        &middot;
        &middot;
        &middot;
        &middot;
    </span>
</template>

<script>
export default {
    props: ['resourceName', 'field'],
}
</script>
