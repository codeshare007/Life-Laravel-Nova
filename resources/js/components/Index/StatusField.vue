<template>
    <div class="flex items-center">
        <span class="mr-3 text-60" v-if="field.loadingWords.includes(field.value)">
            <loader width="30" />
        </span>

        <span :class="{ 'text-danger': field.failedWords.includes(field.value) }">
            {{ field.value }}
        </span>
    </div>
</template>

<script>
export default {
    props: ['resourceName', 'field'],
}
</script>
