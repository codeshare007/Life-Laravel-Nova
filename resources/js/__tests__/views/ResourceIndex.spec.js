import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import Index from '@/views/Index.vue'

// Create a renderer for snapshot testing
const renderer = createRenderer()

// Nova global mock
// class Nova {
//     constructor(config) {}

//     request() {
//         return {
//             get() {
//                 return { data: {} }
//             },
//         }
//     }
// }

// global.Nova = new Nova()

describe('Index.vue', () => {
    it('renders', () => {
        const wrapper = shallowMount(Index, {
            stubs: ['loading-view', 'cards'],
            propsData: {
                resourceName: 'posts',
            },
        })
        renderer.renderToString(wrapper.vm, (err, str) => {
            if (err) throw new Error(err)
            expect(str).toMatchSnapshot()
        })
    })

    it('renders after loading', () => {
        const wrapper = shallowMount(Index, {
            stubs: ['loading-view', 'cards'],
            propsData: {
                resourceName: 'posts',
            },
        })

        expect(wrapper.vm.initialLoading).toEqual(false)
    })

    it('should show its cards', () => {
        const $route = { params: { resourceName: 'posts' } }
        const wrapper = shallowMount(Index, {
            stubs: ['loading-view', 'cards'],
            mocks: {
                $route,
            },
            propsData: {
                resourceName: 'posts',
            },
        })

        // wrapper.setData({
        //     cards: [{}],
        // })

        expect(wrapper.vm.shouldShowCards).toEqual(true)
    })
})
