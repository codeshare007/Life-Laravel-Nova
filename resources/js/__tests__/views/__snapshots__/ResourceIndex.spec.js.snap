// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index.vue renders 1`] = `<loading-view-stub></loading-view-stub>`;
